fenzo-core/src/main/java/com/netflix/fenzo/AssignableVirtualMachine.java

```diff
@@ -397,8 +397,14 @@ String getCurrVMId() {
     boolean addLease(VirtualMachineLease lease) {
         if (logger.isDebugEnabled())
             logger.debug("{}: adding lease id {}", hostname, lease.getId());
-        if(singleLeaseMode && firstLeaseAdded)
-            return false;
+        if(singleLeaseMode && firstLeaseAdded) {
+            if (leasesMap.isEmpty()) {
+                leasesMap.put(lease.getId(), lease);
+                return true;

```

Comment:
Might be better to replace the old lease with with a new one.

Reply:
It used to work this way so am going to leave it the same way. However, the expectation in single lease mode is that the leases are the same.